hydromt/data_adapter/dataset.py

```diff
@@ -0,0 +1,431 @@
+            - 'source_url'
+            - 'source_license'
+            - 'paper_ref'
+            - 'paper_doi'
+            - 'category'
+        placeholders: dict, optional
+            Placeholders to expand yaml entry to multiple entries (name and path)
+            based on placeholder values
+        attrs: dict, optional
+            Additional attributes relating to data variables. For instance unit
+            or long name of the variable.
+        driver_kwargs, dict, optional
+            Additional key-word arguments passed to the driver.
+        storage_options: dict, optional
+            Additional key-word arguments passed to the fsspec FileSystem object.
+        name, catalog_name: str, optional
+            Name of the dataset and catalog, optional.
+        provider: str, optional
+            A name to identifiy the specific provider of the dataset requested.
+            if None is provided, the last added source will be used.
+        version: str, optional
+            A name to identifiy the specific version of the dataset requested.
+            if None is provided, the last added source will be used.
+        """
+        super().__init__(
+            path=path,
+            driver=driver,
+            filesystem=filesystem,
+            nodata=nodata,
+            rename=rename,
+            unit_mult=unit_mult,
+            unit_add=unit_add,
+            meta=meta,
+            attrs=attrs,
+            driver_kwargs=driver_kwargs,
+            storage_options=storage_options,
+            name=name,
+            catalog_name=catalog_name,
+            provider=provider,
+            version=version,
+        )
+
+    def to_file(
+        self,
+        data_root: str | Path,
+        data_name: str,
+        time_tuple: Optional[Tuple[str | datetime]] = None,
+        variables: Optional[List[str]] = None,
+        driver: Optional[str] = None,
+        **kwargs,
+    ) -> Tuple[str, str]:
+        """Save a dataset slice to file.
+
+        Parameters
+        ----------
+        data_root : str, Path
+            Path to output folder
+        data_name : str
+            Name of output file without extension.
+        variables : list of str, optional
+            Names of Dataset variables to return. By default all dataset variables
+            are returned.
+        time_tuple : tuple of str, datetime, optional
+            Start and end date of period of interest. By default the entire time period
+            of the dataset is returned.
+        driver : str, optional
+            Driver to write file, e.g.: 'netcdf', 'zarr', by default None
+        **kwargs
+            Additional keyword arguments that are passed to the `to_zarr`
+            function.
+
+        Returns
+        -------
+        fn_out: str
+            Absolute path to output file
+        driver: str
+            Name of driver to read data with, see
+            :py:func:`~hydromt.data_catalog.DataCatalog.get_dataset`
```

Comment:
I actually like this, but I'm not sure if the other adapters do this? If not we can make a ticket to add that later, but it's a nice pattern for sure.